Nos/Views/Modifiers/ReportMenuModifier.swift

```diff
@@ -173,16 +195,26 @@ struct ReportMenuModifier: ViewModifier {
         }
     }
     
-    func mute(author: Author) {
-        Task {
-            do {
-                try await author.mute(viewContext: viewContext)
-            } catch {
-                Log.error(error.localizedDescription)
+    func mute(author: Author) async {
+        do {
+            try await author.mute(viewContext: viewContext)
+        } catch {
+            Log.error(error.localizedDescription)
+        }
+    }
+
+    /// Determines the visibility status for a flagged account and applies the appropriate action.
+    private func determineFlaggedAccoutVisibility() async {
```

Comment:
Now that this doesn't unmute, I might name it `muteIfNeeded` or `muteAccountIfNeeded` or `muteUserIfNeeded`. But it's not a big deal.